packages/projection/src/operators/Mappers/withHandles.ts

```diff
@@ -62,6 +64,45 @@ const tryCreateHandleOwnership = (
   }
 };
 
+const getHandleMetadata = (handleDataFields: Cardano.PlutusList, logger: Logger) =>
+  handleDataFields.items.map((data) => {
+    if (Cardano.util.isPlutusMap(data)) {
+      return Cardano.util.tryConvertPlutusMapToUtf8Record(data, logger);
+    }
+  });
+
+const getParentHandleFromResolvedAddresses = (resolvedAddresses: Cardano.PlutusMap, logger: Logger) => {
+  const decodedResolvedAddresses = Cardano.util.tryConvertPlutusMapToUtf8Record(resolvedAddresses, logger);
+
+  if (!Cardano.util.isPlutusMap(decodedResolvedAddresses) && decodedResolvedAddresses.ada instanceof Uint8Array) {
+    return Cardano.Address.fromBytes(HexBlob.fromBytes(decodedResolvedAddresses.ada)).toBech32();
+  }
+};
+
+const getHandleParentAddress = (handleData: HandleOwnership, logger: Logger) => {
+  if (handleData.datum && Cardano.util.isConstrPlutusData(handleData.datum)) {
+    const metadata = getHandleMetadata(handleData.datum.fields, logger);
+
+    const resolvedAddressEntry = metadata.find(
+      (metadataEntry) => typeof metadataEntry?.resolved_addresses !== 'undefined'
+    );
+
+    if (
+      resolvedAddressEntry?.resolved_addresses &&
+      Cardano.util.isPlutusMap(resolvedAddressEntry?.resolved_addresses)
+    ) {
+      // This is a virtual subhandle. We get the parent handle from the resolved addresses.
```

Comment:
Virtual handle `resolved_addresses` is the owner of the virtual handle (handle resolves to that address): `HandleOwnership.latestOwnerAddress`. Parent handle is not being resolved here